### ISSUE TITLE
feature: Async files uploading

### ISSUE THREAD

**Issue by SatarisGIT** [uxbox/uxbox#3014]
### Is your feature request related to a problem? Please describe.

Hello everyone! I have some optimization proposal

So - when you upload for example 1000 images on PenPot in "Assets/Graphic" section, every item starts upload after previous fully uploaded. You can see this on waterfall screenshot

![image](https://user-images.githubusercontent.com/28608081/223150254-231a8e0f-93b0-47a0-9c21-7be821bc172c.png)

It would be nice when this operation will be asynchronous instead of synchronous. I think it is very easy to implements and when you create some large design systems it would be super nice to have.

### Describe the solution you'd like.

Use async methods on upload multiple files on frontend

**Comment by niwinz** [comment]
Hello @SatarisGIT 

I'm not clearly understand what you have in mind when you say asynchronous. Because making asynchronous for the user means you don't need to block the user interaction on the upload process. But this has nothing todo with the waterfall; waterfall is because the assets are uploaded one after other, serially. Maybe you intend to say in parallel?

**Comment by SatarisGIT** [comment]
Yeah, parallel, when you start uploading for example 100 files - 100 files may start uploading in same time, not waiting for others

**Comment by niwinz** [comment]
Yep, we have considered it but we still prefer the current approach because not everybody has a huge 100M-1G fiber optic bandwidth... starting 100 uploads in parallel will not make upload faster magically if you don't have huge bandwidht, in contrast it will make it a bit slower.
We also need to consider the server resource usage allocated to just one user uploading 100 files or more.
Of course, it can be improved probably in batches of 2 or so, but starting uploading 100 or more files in parallel has too many implications and we not going to activate, sorry.

In any case, thanks for the suggestion.
